src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const statesArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        statesArray.push({ ...stateClone });
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateClone[key];
+        }
+        statesArray.push({ ...stateClone });
+        break;
+      case 'clear':
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
+        statesArray.push({ ...stateClone });
+        break;
+    }
```

Comment:
[CODE STYLE]: A `default` case should be included in your `switch` statement for error handling. This ensures that there's a fallback for any action types that aren't recognized.